docs/extensions/azure.md

```diff
@@ -205,3 +245,93 @@ Where `variable_name` can be one of the following:
+├── l_receipmonth=1997-11
+│   ├── l_shipmode=AIR
+│   │   └── data_0.csv
+│   ├── l_shipmode=SHIP
+│   │   └── data_0.csv
+│   └── l_shipmode=TRUCK
+│       └── data_0.csv
+└── l_receipmonth=1997-12
+    ├── l_shipmode=AIR
+    │   └── data_0.csv
+    ├── l_shipmode=SHIP
+    │   └── data_0.csv
+    └── l_shipmode=TRUCK
+        └── data_0.csv
+```
+
+The query performed through the blob endpoint
+
+```sql
+SELECT count(*)
+FROM 'az://root/l_receipmonth=1997-*/l_shipmode=SHIP/*.csv';
+```
+
+Will perform the following steps:
+
+* List all the files with the prefix `root/l_receipmonth=1997-`
+  * root/l_receipmonth=1997-10/l_shipmode=SHIP/data_0.csv
+  * root/l_receipmonth=1997-10/l_shipmode=AIR/data_0.csv
+  * root/l_receipmonth=1997-10/l_shipmode=TRUCK/data_0.csv
+  * root/l_receipmonth=1997-11/l_shipmode=SHIP/data_0.csv
+  * root/l_receipmonth=1997-11/l_shipmode=AIR/data_0.csv
+  * root/l_receipmonth=1997-11/l_shipmode=TRUCK/data_0.csv
+  * root/l_receipmonth=1997-12/l_shipmode=SHIP/data_0.csv
+  * root/l_receipmonth=1997-12/l_shipmode=AIR/data_0.csv
+  * root/l_receipmonth=1997-12/l_shipmode=TRUCK/data_0.csv
+* Filter the result with the requested pattern `root/l_receipmonth=1997-*/l_shipmode=SHIP/*.csv`
+  * root/l_receipmonth=1997-10/l_shipmode=SHIP/data_0.csv
+  * root/l_receipmonth=1997-11/l_shipmode=SHIP/data_0.csv
+  * root/l_receipmonth=1997-12/l_shipmode=SHIP/data_0.csv
+
+The same query performed through the datalake endpoint:
+
+```sql
+SELECT count(*)
+FROM 'abfss://root/l_receipmonth=1997-*/l_shipmode=SHIP/*.csv';
+```
+
+Will perform the following steps:
+
+* List all directories in `root/`
+  * root/l_receipmonth=1997-10
+  * root/l_receipmonth=1997-11
+  * root/l_receipmonth=1997-12
+* Filter and list subdirectories: `root/l_receipmonth=1997-10`, `root/l_receipmonth=1997-11`, `root/l_receipmonth=1997-12`
+  * root/l_receipmonth=1997-10/l_shipmode=SHIP
+  * root/l_receipmonth=1997-10/l_shipmode=AIR
+  * root/l_receipmonth=1997-10/l_shipmode=TRUCK
+  * root/l_receipmonth=1997-11/l_shipmode=SHIP
+  * root/l_receipmonth=1997-11/l_shipmode=AIR
+  * root/l_receipmonth=1997-11/l_shipmode=TRUCK
+  * root/l_receipmonth=1997-12/l_shipmode=SHIP
+  * root/l_receipmonth=1997-12/l_shipmode=AIR
+  * root/l_receipmonth=1997-12/l_shipmode=TRUCK
+* Filter and list subdirectories: `root/l_receipmonth=1997-10/l_shipmode=SHIP`, `root/l_receipmonth=1997-11/l_shipmode=SHIP`, `root/l_receipmonth=1997-12/l_shipmode=SHIP`
+  * root/l_receipmonth=1997-10/l_shipmode=SHIP/data_0.csv
+  * root/l_receipmonth=1997-11/l_shipmode=SHIP/data_0.csv
+  * root/l_receipmonth=1997-12/l_shipmode=SHIP/data_0.csv
+
+As you can see because blob do not support the notion of directories the filter can only be performed after the listing whereas the datalake endpoint will list recursively directory by directory. In the example it does not look that different but with more partitions or more files under the partitions that can make a lot of differences.
```

Comment:
```suggestion
As you can see because the Blob endpoint does not support the notion of directories, the filter can only be performed after the listing, whereas the ADLS endpoint will list files recursively. Especially with higher partition/directory counts, the performance difference can be very significant.
```